tests/test_main.py

```diff
@@ -0,0 +1,38 @@
+import requests
+from fastapi.testclient import TestClient
+from main import app
+
+
+client = TestClient(app)
+
+
+# check that the homepage returns a successful status code
+def test_app():
+    resp = client.get("/")
+    assert resp.raise_for_status() != requests.exceptions.HTTPError
```

Comment:
This is not testing anything. What you need to test is that your api is returning what you expect it to return. For the root, you are just returning a string, so we need to check your API is returning the string successfully, so we need to check for the status 200 and for the content of the response:
```
def test_app():
    resp = client.get("/")

    assert resp.status_code == 200
    assert resp.json() == "Random fact (from a list) generated at '/facts'. Random breed generated at '/breeds'."
```